src/analysis/QueryParser.ts

```diff
@@ -185,42 +191,44 @@ class QueryParser {
          * 2. there is a directive named inlcude and the value is true
          * 3. there is a directive named skip and the value is false
          */
-        // const directive = node.directives;
-        // if (directive && this.directiveCheck(directive[0])) {
-        this.depth += 1;
-        if (this.depth > this.maxDepth) this.maxDepth = this.depth;
-        // the kind of a field node will either be field, fragment spread or inline fragment
-        if (node.kind === Kind.FIELD) {
-            complexity += this.fieldNode(node, parentName.toLowerCase());
-        } else if (node.kind === Kind.FRAGMENT_SPREAD) {
-            // add complexity and depth from fragment cache
-            const { complexity: fragComplexity, depth: fragDepth } =
-                this.fragmentCache[node.name.value];
-            complexity += fragComplexity;
-            this.depth += fragDepth;
+        const directive = node.directives;
+        if (directive && this.directiveCheck(directive[0])) {
```

Comment:
We should also handle cases where there are multiple directives attached to this node rather than just checking the first directive.

Reply:
I've added this. What is your take on this check:
```javascript
if (node.directives && !this.directiveExcludeField([...node.directives])) {
    //code
}
```

`node.directives` is always truthy but TS requires the check in order to access the array value at `node.directives`.